### PR TITLE
[Pull Request] ProskillAPI QoL Update - Introducing Default Skills

### PR DESCRIPTION
**Pull request by iomatix**
Created new list for class called "default-skills".
The default skills are skills which level is 1 from start. 
The mechanic and changes to the code are really simple but the change is really QoL.

So the new feature is able to create new hidden from eyes of the players passive skills, calculated on raw numbers and saved values or current attributes.
New way for creating passives:
```
  skills: 
  - 'New_Passive_1'
  - 'New_Passive_2'
  default-skills: 
  - 'New_Passive_1'
  - 'New_Passive_2'
``` 
So `New_Passsive_1` and `New_Passive_2` gonna be at 1st level from beginning (e.g. `/class profess [CLASS_NAME]`) without using SkillPoints.

### CODE REVIEW

File: src/main/java/com/sucy/skill/api/player/PlayerData.java [hunk 1]
```diff
@@ -1634,6 +1634,11 @@ public boolean profess(RPGClass rpgClass) {
                 if (!skills.containsKey(skill.getKey())) {
                     skills.put(skill.getKey(), new PlayerSkill(this, skill, current));
                     combos.addSkill(skill);
+
+                    // iomatix: check default skills and upgrade to 1 if any (Upgraded to 1 by default)
+                    if (rpgClass.getDefaultSkills(!isResetting).contains(skill)){
+                        if (getSkillLevel(skill.getName()) < 1)  forceUpSkill(skills.get(skill.getKey()), 1);
```
Comment (iomatix) [inline src/main/java/com/sucy/skill/api/player/PlayerData.java:1640]: Check if shouldn't be `getSkillLevel(...) == 0`

**Comment by fructozer** [comment]
I don't think this is necessary. ProSkillAPI already has an auto level feature. Just set the default cost to zero, you will have this skill at level 1 when you just receive the class.

**Comment by Travja** [comment]
Does this offer any real benefit over setting a skill's cost to zero so the auto-leveling feature kicks in?

**Comment by iomatix** [comment]
> Does this offer any real benefit over setting a skill's cost to zero so the auto-leveling feature kicks in?

Hmm, I wasn't aware of the auto leveling feature, I'll test today if this one works correctly with cost set up to 0. Is it necessary to switch variable in config file?
I did the feature because it didn't kick in for me in situation there was no cast and the skill wasn't visible in the GUI. 
I'll verify the matter today and reply there once again.

**Comment by iomatix** [comment]
> > Does this offer any real benefit over setting a skill's cost to zero so the auto-leveling feature kicks in?
> 
> Hmm, I wasn't aware of the auto leveling feature, I'll test today if this one works correctly with cost set up to 0. Is it necessary to switch variable in config file? I did the feature because it didn't kick in for me in situation there was no cast and the skill wasn't visible in the GUI. I'll verify the matter today and reply there once again.

So I did the skill with these settings and gonna test the default auto-leveling feature.
[The skill settings (image)](https://github.com/promcteam/proskillapi/assets/13110161/b8e456bf-d92f-4bb0-9737-a95ec080fcb1)

**Comment by iomatix** [comment]
> > > Does this offer any real benefit over setting a skill's cost to zero so the auto-leveling feature kicks in?
> > 
> > 
> > Hmm, I wasn't aware of the auto leveling feature, I'll test today if this one works correctly with cost set up to 0. Is it necessary to switch variable in config file? I did the feature because it didn't kick in for me in situation there was no cast and the skill wasn't visible in the GUI. I'll verify the matter today and reply there once again.
> 
> So I did the skill with these settings and gonna test the default auto-leveling feature. [The skill settings (image)](https://github.com/promcteam/proskillapi/assets/13110161/b8e456bf-d92f-4bb0-9737-a95ec080fcb1)

Worked... yeah it's nothing new then. Let's close the pull request.

**Pull request closed without merging**